### DISCUSSION TITLE
MacOS clipboard integration

### DISCUSSION THREAD

**Discussion by iloveitaly** [community/community#33417] [Codespaces]
I'm just getting started with codespaces! What an amazing tool. Great work!

Question for the community:

A common workflow I have is copying from the terminal. I do this via shell aliases/functions using `pbcopy` or using tmux copy mode (in combination with `reattach-to-user-namespace`. Is there a way to support this workflow from codespaces? i.e. insert text from the shell into the macos clipboard?

An idea I had was ssh'ing into the codespace terminal via tmux running locally on my machine, but it's not clear how I would set that up. It seems like there are some commands which 'magically' integrate with macos (`gh pr view --web` for instance opens up a URL in the default macos web browser). I imagine there is a way to do something similar with the clipboard?

**Comment by samruddhikhandale** [comment]
As Linux is the only OS which Codespaces support, have your tried using `pbcopy` alternative for Linux (probably [xclip](https://github.com/astrand/xclip)) ? 

`gh pr view --web` is one of the commands provided by [GitHub CLI ](https://cli.github.com/) which supports Mac OS, hence the magical integration.

**Comment by iloveitaly** [comment]
The trick is getting the clipboard to integrate with the host OS (in my case macos). https://github.com/quackduck/uniclip looks like it could be a solution, but I haven't gotten around to tinkering with it to get it working.
